### DISCUSSION TITLE
Average interpolations are adding slight black opacity to transparent areas on rotate

### DISCUSSION THREAD

**Discussion by jk-keller** [ImageMagick/ImageMagick#7401] [Help] [answered]
### ImageMagick version

7.1.1-33 Q16-HDRI aarch64 22263

### Operating system, version and so on

macOS Sonoma 14.5, MacBook Pro M1 Max

### Description

When rotating a png with transparency, the 'Average' interpolations will turn any transparency in source image into a partly opaque black area (25% for Avg4), then rotate the image, adding a fully transparent background for the rotational extension of canvas. One would expect the source transparency to remain, except at the edges of Mario, where it should be partly transparent red, blue, brown, etc.

`magick 01--mario.png -virtual-pixel transparent -background transparent -interpolate Average4 -rotate 10 01--mario-rotated.png`

![01--mario](https://github.com/ImageMagick/ImageMagick/assets/25354950/4823c174-b1c6-4ed0-8cea-fc6d1b35763d)
source

![01--mario-rotated](https://github.com/ImageMagick/ImageMagick/assets/25354950/a99ba31e-adef-46ac-8e3e-b3cbb075b0c3)
result (_put over light background to see the rgba(0,0,0,.25) rotated rect_)

This works as expected for all the other interpolation methods. I've tried adding all of the `-alpha` options to no help, but feel like this may be a multi-step process, where can't figure out the other steps. What am I missing?

My setup:
`7.1.1-33 Q16-HDRI aarch64 22263` on `macOS Sonoma 14.5, MacBook Pro M1 Max`

**Comment by fmw42** [comment]
I see the same effect on IM 7.1.1.33 Mac OSX Ventura.

**Comment by urban-warrior** [comment, marked as answer]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.

**Reply by jk-keller** [reply]
Working, thank you!
